taipy/core/_scheduler/_scheduler.py

```diff
@@ -195,3 +204,38 @@ def _update_job_config(cls):
             cls._dispatcher = _StandaloneJobDispatcher()
         else:
             cls._dispatcher = _DevelopmentJobDispatcher()
+
+    @classmethod
+    def _set_process_in_scheduler(cls, job_id, process):
+        cls._processes[job_id] = process
+
+    @classmethod
+    def _pop_process_in_scheduler(cls, job_id, default=None):
+        return cls._processes.pop(job_id, default)  # type: ignore
+
+    @classmethod
+    def _cancel_job(cls, job):
+        if process := cls._pop_process_in_scheduler(job.id):
```

Comment:
What about retrieving the submit_id here and cancel all the jobs with the same submit_id.
would it make the code design easier to understand?

Reply:
im not sure about the 1st part "What about retrieving the submit_id here", as for the 2nd part "cancel all the jobs with the same submit_id", we discussed and want to cancel only the subsequent jobs and leave the others to continue running :D